src/Microsoft.TestPlatform.PlatformAbstractions/uap10.0/System/PlatformThread.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.PlatformAbstractions
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.Interfaces;
+
+    public class PlatformThread : IThread
+    {
+        /// <inheritdoc/>
+        public void Run(Action action, PlatformApartmentState apartmentState, bool waitForCompletion)
+        {
+            if (apartmentState == PlatformApartmentState.STA)
```

Comment:
You are right, we are logging the warning and running in MTA in  caller (`BaseRunTests.cs`). As there is no way to log in platform abstraction, we are doing that by throwing exception.